src/history/history-and-comments.component.tsx

```diff
@@ -214,7 +213,7 @@ const HistoryAndComments: React.FC<{
       {medicationRequestBundles &&
         medicationRequestBundles
           .flatMap((medicationDispenseBundle) => medicationDispenseBundle.dispenses)
-          .sort(sortMedicationDispensesByDateRecorded)
+          .sort(sortMedicationDispensesByWhenHandedOver)
```

Comment:
The FHIR resource for a Medication Dispense supports three dates:
* date recorded (maps to date created in the OpenMRS model)
* when prepared (maps to date prepared in the OpenMRS model)
* when handed over (maps to date handed over in the OpenMRS model)

Previously we were timestamping "when prepared" and "when handed over" to "now" (and date_created was timestamped to "now" on the backend).

Now we allow the user to set the "when prepared" and "when handed over" dates via a single Date of Dispense datepicker (which defaults to now).

I don't remember why we were using the date recorded as the display and sort date, but now that we are allowed orders to be back dated, we need to display and sort by when prepared or when handled over, because these are the more accurate and relevant dates.

We are continuing to not distinguish between "date prepared" and "date handed over" and always set them to the  single "dispense date", but the data module will allow us to expand the Dispensing app to record these two potentially distinct events if we like.